### DISCUSSION TITLE
Easy update function for Rust Desk Clients & Server

### DISCUSSION THREAD

**Discussion by breathless19** [rustdesk/rustdesk#3347] [General]
### Description

I'm running Rustdesk Server on Windows 11.    It would be nice to have an easy method to update / autoupdate Rustdesk.exe client  (Rustdesk nightly or regular) and Rustdesk Server (hbbr / hbbs services) without losing any configuration settings.    

I cannot find anywhere in the documentation how this can be done.   

### Impact

Updating the executable of the RustDesk.exe client is easy enough by just replacing it on "my side", but being able to have it auto update on the remote side, or even just a "check for update" option in the client would help ensure that remote clients are always up to date, whether they are using Nightly or not.   

Likewise for the server.  Running the new (I think its new) "Rustdeskserver.setup.exe" doesn't update my currently installed RustDesk service files - it creates  separate service entry's / executable that do not import my current server configuration (and therefore don't work), but rather run parallel to it.    It would be very helpful if this could be done automatically or semi-automatically with an included "UpdateRustdeskServer.exe".

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/3753

**Reply by glomme** [reply]
> #3753
I have read the article, but I would expect that if I have a Pro Server running, that I can also automatically update the custom clients I have created. 
I could also imagine that the client I am accessing can be updated remotely, similar to TeamViewer.
I can neither update the client from the server nor with an existing remote connection right now.
The only thing I can do right now is download the actual client version and update it by hand.

**Reply by derda83** [reply]
> #3753

We were currently evaluating rustdesk server pro for managing a few thousand clients and stumbled upon this missing key feature. Unfortunately, this is a show blocker, as there is no easy way to auto update custom clients - or we cannot find that option.

**Comment by boundedxd** [comment]
I have the same problem, I am gooning and can't find time to update it takes a whole 2 seconds man I hate this

**Comment by AndreiCherniaev** [comment]
I found this on [SO](https://stackoverflow.com/questions/25923939/how-do-i-download-binary-files-of-a-github-release/79669461#79669461): to download binary files of a GitHub release you can use [binup](https://github.com/KonishchevDmitry/binup) or [hubapp](https://github.com/warrensbox/hubapp) or [eget](https://github.com/zyedidia/eget). Hope it help

**Comment by morno** [comment]
see this https://github.com/rustdesk/rustdesk-server-pro/discussions/661#discussioncomment-16015675 hope that it works as intended :)

https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#allow-auto-update
